src/cdk/overlay/position/global-position-strategy.ts

```diff
@@ -169,7 +169,7 @@ export class GlobalPositionStrategy implements PositionStrategy {
 
     if (config.width === '100%') {
       parentStyles.justifyContent = 'flex-start';
-    } else if (this._overlayRef.getConfig().direction === 'rtl') {
+    } else if (this._overlayRef.getConfig().direction === 'rtl' && this._justifyContent.startsWith('flex-')) {
```

Comment:
Out of context, it's hard to see the reasoning behind this check. I would switch it so the check is `this._overlayRef.getConfig().direction === 'rtl' && this._justifyContent !== 'center'`.

Reply:
Since the overlay wrapper can only have one overlay panel at a time, I wouldn't worry too much about any of the other flex values.

Reply:
I thought about another solution for that:
```javascript
if (config.width === '100%') {
    parentStyles.justifyContent = 'flex-start';
} else if (this._overlayRef.getConfig().direction === 'rtl') {
    // In RTL the browser will invert `flex-start` and `flex-end` automatically, but we
    // don't want that because our positioning is explicitly `left` and `right`, hence
    // why we do another inversion to ensure that the overlay stays in the same position.
    // TODO: reconsider this if we add `start` and `end` methods.
    if (this._justifyContent === 'flex-start') {
        parentStyles.justifyContent = 'flex-end';
    } else if (this._justifyContent === 'flex-end') {
        parentStyles.justifyContent = 'flex-start';
    }
}

parentStyles.justifyContent = parentStyles.justifyContent || this._justifyContent;
```
Or maybe change the last line to:
```javascript
if (config.width === '100%') {
    parentStyles.justifyContent = 'flex-start';
} else if (this._overlayRef.getConfig().direction === 'rtl') {
    // In RTL the browser will invert `flex-start` and `flex-end` automatically, but we
    // don't want that because our positioning is explicitly `left` and `right`, hence
    // why we do another inversion to ensure that the overlay stays in the same position.
    // TODO: reconsider this if we add `start` and `end` methods.
    if (this._justifyContent === 'flex-start') {
        parentStyles.justifyContent = 'flex-end';
    } else if (this._justifyContent === 'flex-end') {
        parentStyles.justifyContent = 'flex-start';
    }
}

if (!parentStyles.justifyContent) {
    parentStyles.justifyContent = this._justifyContent;
}
```
What do you think?

Reply:
The latter option looks good as well.